src/core_modules/capture-core/components/WidgetsChangelog/common/hooks/useChangelogData.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+import moment from 'moment';
+import { v4 as uuid } from 'uuid';
+import { useMemo, useState } from 'react';
+import { useTimeZoneConversion } from '@dhis2/app-runtime';
+import { useApiDataQuery } from '../../../../utils/reactQueryHelpers';
+import { CHANGELOG_ENTITY_TYPES, QUERY_KEYS_BY_ENTITY_TYPE } from '../Changelog/Changelog.constants';
+import type { Change, ChangelogRecord, ItemDefinitions, SortDirection } from '../Changelog/Changelog.types';
+import { convertServerToClient } from '../../../../converters';
+import { convert } from '../../../../converters/clientToList';
+
+type Props = {
+    entityId: string,
+    programId?: string,
+    entityType: $Values<typeof CHANGELOG_ENTITY_TYPES>,
+    dataItemDefinitions: ItemDefinitions,
+}
+
+const DEFAULT_PAGE_SIZE = 10;
+const DEFAULT_SORT_DIRECTION = 'default';
+
+const getMetadataItemDefinition = (
+    elementKey: string,
+    change: Change,
+    dataItemDefinitions: ItemDefinitions,
+) => {
+    const { dataElement, trackedEntityAttribute } = change;
+    const fieldId = dataElement ?? trackedEntityAttribute;
+    const metadataElement = fieldId ? dataItemDefinitions[fieldId] : dataItemDefinitions[elementKey];
+
+    return { metadataElement, fieldId };
+};
+
+export const useChangelogData = ({
+    entityId,
+    entityType,
+    dataItemDefinitions,
```

Comment:
You might still be waiting for the backend here, but we will have to put the `programId` in here.

Reply:
.. and use it in the request